### ISSUE TITLE
Add an option to simulate retroactive Endurance

### ISSUE THREAD

**Issue by Cozzolino92** [robot9706/FSSE#167]
One of the biggest issue with this game is the fact that Endurance does not work retroactively and how it twists and dominates the gameplay. Right now, most of us players train Endurance at level 1 and only allow their Dwellers to level up with 10 Endurance. We also gearswap Endurance outfits upon level up to have perfect E17 Dwellers and I think we all agree this little chore adds nothing to gameplay.

I would like to ask for a feature that updates Dwellers Health Pool based on the Endurance formula so to simulate a retroactive Endurance HP scaling. More over, said option should account for the most common gearswapping bonuses (+3,+5,+7). Endurance formula is known and all the possible combinations can be easily calculated and found [here](https://docs.google.com/spreadsheets/d/1b3n_dn3Hbk9UVRyMrnbgoCJTrpVBe1pOSgzL9hjRFoY/edit#gid=0).

The best way to implement this for the users counts two options:

1 - Quick update: reads each Dweller current level and Endurance value, has an option to apply a bonus +3/+5/+7 Endurance, then updates their Health Pool to match retroactive HP growth and gearswapping
2 - Flat update: reads each Dweller current level and updates their Health Pool to match a chosen Endurance value up to E17

The first option preserves gameplay integrity by giving each Dweller the exact Health Pool they would have at their current Endurance, current level, and available Outfit bonuses.
The second option is based on level and disregards the real, current Endurance to flat out update the Health Pool of each Dweller according to a fixed arbitrary Endurance value.

Why do we need this? As already mentioned, gearswapping is a very annoying and time consuming strategy that adds nothing to the gameplay in on itself. On another note, manually editing Health Pools is time consuming aswell and must be repeated over and over until Dwellers hit level 50.

Please, consider this feature, I firmly believe it would make a lot of players happy. Thanks for your effort and time : )

**Comment by robot9706** [comment]
I know that this Endurance leveling thing is really cumbersome, I don't follow the game anymore but I've heard a lot about it, even when I used to play it.

You detailed your feature proposal really well, but I have one question. Basically I don't understand the +3/5/7 thing in the "Quick update" feature, could you elaborate why would you need those 3 specific values?

Also just to sum up and see if I understood you correctly:
- Quick update: Grab the Endurance (add some bonus?) and the Level of each dweller and lookup the correct max HP from the spreadsheet.
- Flat update: Grab the Level of each dweller and assume E17 then lookup the max HP value from the spreadsheet.
So basically they do the exact same thing, except te Endurance is different.

**Comment by Cozzolino92** [comment]
I understand how asking for those 3 _specific_ bonus values can be a bit puzzling, I'll try to explain the reason behind this.

I assume you are familiar with the fact that Endurance is not retroactive, I also assume you are aware that outfits with Endurance bonuses can break the 10 Endurance barrier and it does count for HP scaling on level up. If you want to have the best possible Dwellers, you are going to have to equip Outfits with Endurance bonuses for each and every level up, and I call that action "gearswapping". The most common outfit you get throughout the game - per each rarity level - give +3/+5/+7 Endurance bonus.

The reason behind this specific request is that I want to somehow preserve the integrity and slow, yet rewarding, pace of the game while "cheating out" one of the most frustrating aspect of it. In my eyes, I'm not asking for a cheat per se, I think this small feature on its own can make the game better and not simply bend its gameplay in the way that a straight "Max stat/hp/level/special" does.

About the requests:

> So basically they do the exact same thing, except te Endurance is different.

Yes, let me get into details in light of the previous explanation:

With the "**Quick update**" feature, your Dwellers _real_ Endurance is taken (and an optional +3/+5/+7 bonus applied - this simulates gearswapping), along with the level, to calculate and update the Health Pool of Dwellers as if Endurance was retroactive. Using this option, a Dweller with 6 Endurance will still have less HP than a Dweller with 10 Endurance, although both will have a retroactive Endurance Health Pool plus any potential Outfit bonus on top of it.

With the "**Flat update**" feature, your Dwellers _real_ Endurance is not a factor, only level is, they all get a flat update to their Health Pool corresponding to an arbitrary selected Endurance value.

As it's been a long time you haven't been onto this, I'll remind you, in case you forgot, that all Dwellers at Level 1 start with 105 HP regardless of their Endurance. Just throwing this out in case you didn't notice it in the spreadsheet.

Again, thanks for your time : )

**Comment by robot9706** [comment]
I attached v1.5.9 of the editor with the feature you asked, it's in the "Tools" tab on the right.

I checked the new max HP values to see if it matches the spreadsheet, and it seems right, but could you check it out, test it a little bit?
[FSSEWin-1.5.9-preview.zip](https://github.com/robot9706/FSSE/files/3038286/FSSEWin-1.5.9-preview.zip)

**Comment by Cozzolino92** [comment]
That was quick! I'll be testing this asap and I'll let you know if everything is fine :+1: 

Edit: My 80 Population vault has been flawlessly update to the correct HP values simulating both retroactive Endurance growth and gearswapping, I tested with and without bonus Endurance. The option for flat health update also works as intended. 

If you're ok with it, I shall open a thread on the FoS subreddit to advertise this feature on your editor

I'm really grateful for this, you have my gratitude and, I'm convinced, the gratitude of all who will use this : )

**Comment by robot9706** [comment]
Sorry for the late response, I didn't see you edited your comment :)

Yeah opening a thread on reddit is fine with me, I saw that you already did that 👍 

If everything is fine, then I'll implement this feature in the Android version too and release them as new versions so users don't need to come to this post just to download it.

**Comment by robot9706** [comment]
The 1.X.8 versions are now available to download: https://robot9706.github.io/FSSE/

The preview is 1.5.9 because I advanced the version number too much :D
